romancal/regtest/test_wfi_pipeline.py

```diff
@@ -0,0 +1,144 @@
+    pipeline.log.info('DMS86 MSG: Testing completion of linearity correction in'
+                  'Level 2 image output.......' +
+                  str(model.meta.cal_step.linearity == 'COMPLETE'))
+    assert model.meta.cal_step.linearity == 'COMPLETE'
+    pipeline.log.info('Status of the step:             ramp_fit      ' +
+                  str(model.meta.cal_step.ramp_fit))
+    pipeline.log.info('DMS86 MSG: Testing completion of ramp fitting in'
+                  'Level 2 image output.......' +
+                  str(model.meta.cal_step.ramp_fit == 'COMPLETE'))
+    assert model.meta.cal_step.ramp_fit == 'COMPLETE'
+    pipeline.log.info('Status of the step:             saturation    ' +
+                  str(model.meta.cal_step.saturation))
+    pipeline.log.info('DMS86 MSG: Testing completion of saturation detection in'
+                  'Level 2 image output.......' +
+                  str(model.meta.cal_step.saturation == 'COMPLETE'))
+    assert model.meta.cal_step.saturation == 'COMPLETE'
+
+    # DMS87 data quality tests
+    pipeline.log.info('DMS87 MSG: Testing existence of data quality array (dq) in'
+                  'Level 2 image output.......' +
+                  str("dq" in model.keys()))
+    assert "dq" in model.keys()
+    pipeline.log.info('DMS87 MSG: Testing existence of general error array (err) in'
+                  'Level 2 image output.......' +
+                  str("err" in model.keys()))
+    assert "err" in model.keys()
+    pipeline.log.info('DMS87 MSG: Testing existence of Poisson noise variance array (var_poisson) in'
+                  'Level 2 image output.......' +
+                  str("var_poisson" in model.keys()))
+    assert "var_poisson" in model.keys()
+    pipeline.log.info('DMS87 MSG: Testing existence of read noise variance array (var_rnoise) in'
+                  'Level 2 image output.......' +
+                  str("var_rnoise" in model.keys()))
+    assert "var_rnoise" in model.keys()
+    pipeline.log.info('DMS87 MSG: Testing existence of flatfield uncertainty variance array (var_flat) in'
+                  'Level 2 image output.......' +
+                  str("var_flat" in model.keys()))
+    assert "var_flat" in model.keys()
+
+    # DMS88 total exposure time test
+    pipeline.log.info('DMS88 MSG: Testing existence of total exposure time (exposure_time)'
+                  'in Level 2 image output.......' +
+                  str("exposure_time" in model.meta.exposure))
+    assert "exposure_time" in model.meta.exposure
+
+    # DMS89 WCS tests
+    pipeline.log.info('DMS89 MSG: Testing that the wcs bounding'
+                  'box was generated.......' +
+                  str(((len(model.meta.wcs.bounding_box) == 2) and
+                        (type(model.meta.wcs.bounding_box[0][0]) == float))))
+    assert ((len(model.meta.wcs.bounding_box) == 2) and
+            (type(model.meta.wcs.bounding_box[0][0]) == float))
```

Comment:
If checking for the type is not in the requirements I would skip it, as there's no reason this should be float, though it usually is. If we need to check the type, I would check it is a numerical type or a scalar.

Reply:
Removed.